Replicator/Worker.hh

```diff
@@ -135,8 +135,12 @@ namespace litecore { namespace repl {
         // Add the token for collection info to the format string
         static inline const char* formatWithCollection(const char* fmt) {
             const std::string fmtStr = format("%s %s", kCollectionLogFormat, fmt);
-            const auto        found  = _formatCache.insert({fmtStr, 0});
-            return found.first->first.data();
+            const auto        found  = _formatCache.find(fmtStr);
```

Comment:
I didn't follow the discussion about the performance but the update is not correct.

This is not thread-safe. There could be one thread inserting and another thread finding. 

This is a double work? I understand that insert() will also check if the item exists before insert.

Reply:
It is _technically_ thread-safe, as an insert will not immediately invalidate iterators (ie the one returned from `find()`). `insert` can invalidate existing iterators if it causes a rehash (capacity of the `set` is exceeded by the `insert` operation). I am improving this thread-safety in the commit I am about to push so it will be 100% thread-safe.

And yes, there is not really any way to do it (and be thread-safe) without doing a `find` and an `insert`, but the extra work done is minimal, as `find` is extremely fast in an `unordered_set` (average O(1) complexity).